app/scripts/controllers/swaps.js

```diff
@@ -187,15 +183,24 @@ export default class SwapsController {
         this.setSwapsErrorKey(QUOTES_NOT_AVAILABLE_ERROR)
       } else {
         const {
-          topAggId: topAggIdAfterGasEstimates,
-          isBest: isBestAfterGasEstimates,
+          topAggId: topAggIdAfterModifications,
+          isBest: isBestAfterModifications,
         } = await this._findTopQuoteAggId(newQuotes)
-        topAggId = topAggIdAfterGasEstimates
-        if (isBestAfterGasEstimates) {
+        topAggId = topAggIdAfterModifications
+        isBest = isBestAfterModifications
+        if (isBestAfterModifications) {
           newQuotes = mapValues(newQuotes, (quote) => ({ ...quote, isBestQuote: false }))
           newQuotes[topAggId].isBestQuote = true
         }
```

Comment:
These lines are now redundant; this is the sole assignment of `isBestQuote` now, so we don't need to un-assign it, and `isBestQuote` is also set below on line 203.

```suggestion
```

Reply:
Or, even better, close this `!approvalRequired && !fetchParams?.balanceError` block after line 184  instead, and have a single call to `_findTopQuoteAggId`. This can all be simplified now.

Reply:
Oh, @rekmarks is way ahead of me I see :sweat_smile:  See #9536

Reply:
I have merged #9536, so this is now resolved